src/pages/vaults/VaultDepositWithdrawForm.tsx

```diff
@@ -567,23 +590,25 @@ export const VaultDepositWithdrawForm = ({
       <$FlexFill />
 
       <WithDetailsReceipt detailItems={inputFormConfig.receiptItems}>
-        <Button
-          type={ButtonType.Submit}
-          action={ButtonAction.Primary}
-          state={{
-            isDisabled: hasInputErrors || !!isAccountViewOnly || !canViewAccount || isSubmitting,
-            isLoading: isSubmitting,
-          }}
-          slotLeft={
-            errorsPreventingSubmit.find((f) => !lightErrorKeys.has(f.code)) != null ? (
-              <$WarningIcon iconName={IconName.Warning} />
-            ) : undefined
-          }
-        >
-          {hasInputErrors && errorsPreventingSubmit[0]?.short != null
-            ? errorsPreventingSubmit[0]?.short
-            : inputFormConfig.buttonLabel}
-        </Button>
+        {maybeConnectWalletButton ?? (
```

Comment:
seems more readable to jsut combine these into a ternary??

```suggestion
!canAccountTrade ? (
    <OnboardingTriggerButton size={ButtonSize.Base} />
  ) : <Button ...;
```

Reply:
The idea was to have the detection logic for the button in one place, then use it twice, as you noted below.